exercises/concept-wip/crystal-hunter/.docs/introduction.md

```diff
@@ -0,0 +1,116 @@
+# Bools
+
+Crystal has a type know as [`Bool`][bools], it is used to represent the values `true` and `false`.
+With the bool type can you use logical operators(`!`, `||`, `&&`) which can be used to combine bools and change their values.
+
+```crystal
+true && true
+# => True
+
+true && false
+# => False
+
+false && false
+# => false
+
+false || true
+# => True
+false || false
+# => false
+
+!false
+# => true
+
+!true
+# => false
+```
+
+## Logical operators
+
+Crystal has 3 logical operators which is used to combine bools and change their values.
+
+### And(`&&`)
+
+The [and operator][and] in crystal is represented by `&&` and returns `true` if both values given is `true` otherwise it returns `false`.
+When using the and operator so should one bool be placed one the right side of the `&&` and one on the left side.
+
+```crystal
+true && true
+# => true
+
+true && false
+# => false
+
+false && false
+# => false
+```
+
+### Or(`||`)
+
+The [or operator][or] in crystal is represented by `||` and returns `true` if **at least one** of values given is `true` if both of the values are `false` then it returns `false`.
+When using the or operator so should one bool be placed one the right side of the `||` and one on the left side.
+
+```crystal
+true || true
+# => true
+
+true || false
+# => true
+
+false || true
+# => true
+
+false || false
+# => false
+```
+
+### Not(`!`)
+
+The not operator in crystal is represented by `!` and returns `true` if the given bool is `false` and return `false` if `true` is given.
+When using the not operator so should the bool be placed after the operator (`!`).
+
+```crystal
+!true
+# => false
+
+!false
+# => true
+```
```

Comment:
All of this is a more detailed repetition of the content right after the first h1 heading ("Bool"). I would avoid repetitions and try to keep the introduction as short as possible (while still giving the students all the knowledge they need to solve the exercise) because people's patience towards reading is very short on the internet 🙂

Reply:
I decided to keep the first part but shorten the example there to make the first part shorter. I hope that improves it.

Reply:
Honestly, I still think the documents (about/introductions) are too long and too detailed. The way you have written them, they are great for somebody that has never heard about logical operators and is learning programming for the first time. But that is explicitly not the target group of Exercism. Exercism is targeting people who can already code in one language and want to learn another. For this reason, concept introductions should mention details that are unique to the programing language, but don't need to explain basic programming concepts that appear in every programming language.

Reply:
I mean it is a valid point, you know if you are a ruby developer, I would tell you to skip this concept because this doesn't teach anything new. The only difference between ruby and crystal is: Crystal doesn't have `and` and `or`. Ruby sees `false` and `nil` as false. I think Crystal sees `false`, `nil`, and `Nop` as false. I am a bit unsure about `Nop`, since it has to do with the compiler and is nothing you actually deal with in normal programming. These differences are kinda minor.

I guess a problem for me is that "Exercism is targeting people who can already code in one language", say a language does something totally different and the way Crystal may do it is "normal" for most languages but not in the perspective of that language. And sure I know around 8 languages, but all of them are quite similar, sure they all have some unique parts, but the most "out-pointing" languages are Elixir and Javascript, Javascript because weakly typed, and Elixir because functionally oriented. It feels hard to give a good perspective if you get what I mean.

I guess the whole explaining part under parentheses could be dropped and under `or` and `and` go to 2 examples instead of 3/4. Do you think such a change is satisfactory?